### ISSUE TITLE
Audio device volume capability on Macs

### ISSUE THREAD

**Issue by eofster** [pjsip/pjproject#2445]
On Macs, when audio device volume capability is supported, it confuses the users. The output volume is remembered during the first call and is reset to the same level for the following calls. The user may adjust the volume in between calls, but at the next call it will be suddenly reset to the level which the device had during the very first call. People don’t understand this behavior and think that the volume changes randomly.

What is the expected behavior of the audio device volume capability? In my experience, people expect the volume to not change automatically. They expect the volume to be exactly at the level they manually set it to before.

**Comment by sauwming** [comment]
If the `keep` parameter is non-zero, it is the expected behaviour indeed, as described in:
https://www.pjsip.org/docs/latest/pjsip/docs/html/group__PJSUA__LIB__MEDIA.htm#ga80e875e32ccc90dc10dd623355aac129

So you may want to set keep to PJ_FALSE if you don't want that.

Let us know if the behaviour is not as specified in the doc.

**Comment by eofster** [comment]
It seems to behave as I described even if the `keep` parameter is set to `PJ_FALSE`. If `keep` is `PJ_TRUE`, it keeps the value specified in `pjsua_snd_set_setting()`. But by default, it keeps the value obtained during the first call.

The capability seems to be enabled from the getter here:
https://github.com/pjsip/pjproject/blob/c2d7c7616fc5240d7c6d56c454ba8f0acf9ac438/pjmedia/src/pjmedia-audiodev/coreaudio_dev.m#L1809
And saved for future use here:
https://github.com/pjsip/pjproject/blob/c2d7c7616fc5240d7c6d56c454ba8f0acf9ac438/pjsip/src/pjsua-lib/pjsua_aud.c#L1736
Here’s the call stack:
```
ca_stream_get_cap
ca_stream_get_param
pjmedia_aud_stream_get_param
update_initial_aud_param
open_snd_dev
pjsua_set_snd_dev2
pjsua_set_snd_dev
pjsua_call_make_call
```

**Comment by gbfarah** [comment]
With reference to above mentioned thread,  I really would like to have an easy was to disable the behaviour in pjsip that is lowering the volume of other applications. I have made the changes in https://github.com/64characters/Telephone/commit/42286e6a74179680af206bc4b8627a803cd5cef3  specifically to the core_audio_dev.m and the volume changing behaviour still remains.

I created this bug https://github.com/pjsip/pjproject/issues/2464 of unusual behavior I am encountering on Catalina version of OSX. Any feedback would be greatly appreciated
